### PR TITLE
add Repoze.who-friendlyform

### PR DESCRIPTION
**Pull request by rluria14**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by rluria14** [comment]
@oblute @mwilson8 agree to be maintainers?

File: recipes/repoze.who-friendlyform/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+    - setuptools
+
+  run:
+    - python
+    - setuptools
```
Comment (marcelotrevisani) [inline recipes/repoze.who-friendlyform/meta.yaml]: Just to double-check.. Do you really need ``setuptools`` to run this project?

File: recipes/repoze.who-friendlyform/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - webob >=0.9.7
+
+test:
+  imports:
+    - repoze.who.plugins.friendlyform
+
+  requires:
+    - repoze.who >=1.0
+    - nose
+    - coverage
```
Comment (marcelotrevisani) [inline recipes/repoze.who-friendlyform/meta.yaml]: It seems that you don't need the ``requires`` as you are just testing the import

File: recipes/repoze.who-friendlyform/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 77138a94b11e8f846889fd7e025d6914e5966125869f6d75ef216801e29cdf13
+  patches:
+    - patch3.patch
```
Comment (marcelotrevisani) [inline recipes/repoze.who-friendlyform/meta.yaml]: Could you please add a comment here explaining the reason for this patch?

Comment (mwilson8) [reply recipes/repoze.who-friendlyform/meta.yaml]: There were imports that were for only for python 2.x (importing from urlparse & urllib) as well as use of a deprecated function from zope.interface

The patch imports from the python 3+ library (urllib.parse) if the python 2.x import fails, and uses an updated function from zope.interface that is backwards compatible

File: recipes/repoze.who-friendlyform/meta.yaml (hunk 1 above)
Comment (rluria14) [reply recipes/repoze.who-friendlyform/meta.yaml]: setup tools removed.

File: recipes/repoze.who-friendlyform/meta.yaml (hunk 2 above)
Comment (rluria14) [reply recipes/repoze.who-friendlyform/meta.yaml]: test requirements removed.

**Comment by rluria14** [comment]
@marcelotrevisani ready for final review and merge

**Pull request merged**